Misc/Adam.cpp

```diff
@@ -2673,47 +2674,176 @@ int IsGridDotted(COMMAND_T* = NULL)
 	return r < 0.000001 || r > 0.66666;
 }
 
+int IsGridSwing(COMMAND_T* = NULL)
+{
+	return GetToggleCommandStateEx(0, 42304); // Grid: Toggle swing grid
+}
+
+int IsAWSetGridPreserveType(COMMAND_T* ct)
+{
+	int toggleState = 0;
+
+	double normalizedGrid = 0;
+	if ((int)ct->user < 0)
+		normalizedGrid = static_cast<double>(abs((int)ct->user)) * 4.0;
+	else
+		normalizedGrid = 4.0 / static_cast<double>((int)ct->user);
+
+	ConfigVar<double> pGridDiv("projgriddiv");
+	if (IsGridTriplet())
+	{
+		if (*pGridDiv == normalizedGrid * (2.0 / 3.0))
+			toggleState = 1;
+	}
+	else if (IsGridDotted())
+	{
+		if (*pGridDiv == normalizedGrid * (3.0 / 2.0))
+			toggleState = 1;
+	}
+	else
+	{
+		if (*pGridDiv == normalizedGrid)
+			toggleState = 1;
+	}
```

Comment:
This can be simplified since all branches are doing the same thing:

```cpp
if (IsGridTriplet())
	normalizedGrid *= 2.0 / 3.0;
else if (IsGridDotted())
	normalizedGrid *= 3.0 / 2.0;

return *pGridDiv == normalizedGrid;
```

Reply:
This is testing for the toggle state, not setting grid division. I really recommend you leave everything I did as I did it because I invested a lot of times into testing it and making sure everything gets properly refreshed everywhere.

Reply:
The refactored code does exactly the same thing as the original. It does not set the grid division either.

These are only improvement suggestions though, I'm not going to edit someone else's PR just like that. 😉

Reply:
You can always edit my stuff man, this is open-source and we're all contributing somehow. I know you know your stuff so I really don't mind. Just please make sure to test any changes, I really had unexpected amount of work to have all of this working properly due to all "refresh" things because you gotta refresh both the arrange and MIDI editor section